bpfd/src/utils.rs

```diff
@@ -23,21 +23,34 @@ pub(crate) fn get_ifindex(iface: &str) -> Result<u32, BpfdError> {
     }
 }
 
-pub(crate) async fn set_map_permissions(directory: &str) {
+pub(crate) async fn set_dir_permissions(directory: &str, mode: u32) {
     // Determine if User Group exists, if not, do nothing
     if get_group_by_name(USRGRP_BPFD).is_some() {
         // Iterate through the files in the provided directory
         for file in fs::read_dir(directory).unwrap() {
-            // Set the permissions on the map such that members of the "bpfd" group can read/write
+            // Set the permissions on the file based on input
```

Comment:
I think we could change the content of this for loop to directly call `set_file_permissions`, it would re-check `USRGRP_BPFD`, but I don't think that's a big deal, WDYT?

```rs
        for file in fs::read_dir(directory).unwrap() {
            match file {
                Ok(file) => set_file_permissions(file.path().to_str().unwrap(), mode).await,
                Err(e) => warn!("Failed to read file: {e}"),
            }
        }
```